api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -405,6 +365,43 @@ private DataSetRequest generateDataSetRequestFromPreviewRequest(
     return ResponseEntity.ok(previewQueryResponse);
   }
 
+  private void addFieldValuesFromBigQueryToPreviewList(List<DataSetPreviewValueList> valuePreviewList, FieldValueList fieldValueList) {
+    IntStream.range(0, fieldValueList.size())
+        .forEach(
+            columnNumber -> {
+              valuePreviewList
+                  .get(columnNumber)
+                  .addQueryValueItem(
+                      Optional.ofNullable(fieldValueList.get(columnNumber).getValue())
+                          .orElse("")
+                          .toString());
```

Comment:
For clarity, the `toString()` should go probably immediately after `getValue()` inside of the `ofNullable` expression, so that the optional expression will be a string in every case. (The empty string in the orElse doesn't need to be converted).

I'm surprised it also works the way it is, since it's an `Optional<Object>`